src/machi_file_proxy.erl

```diff
@@ -529,20 +578,27 @@ check_or_make_tagged_csum(OtherTag, _ClientCsum, _Data) ->
 % </li>
 %
 % On success, `{ok, Bytes, Checksum}' is returned.
-handle_read(FHd, Filename, undefined, Offset, Size, U) ->
-    handle_read(FHd, Filename, machi_util:make_tagged_csum(none), Offset, Size, U);
-
-handle_read(FHd, Filename, TaggedCsum, Offset, Size, U) ->
-    case is_byte_range_unwritten(Offset, Size, U) of
-        true ->
+handle_read(FHd, Filename, CsumTable, Offset, Size, U) ->
+    case {is_byte_range_unwritten(Offset, Size, U),
+          machi_csum_table:find(CsumTable, Offset, Size)} of
+        {true, _} ->
+            %% More close look might be needed to care result of csum
+            %% table, but for now this just works for known chunks
+            %% (TODO^^;).
             {error, not_written};
-        false ->
+        {false, {error, trimmed} = E} ->
+            E;
+        {false, {ok, TaggedCsum}} ->
+            %% Something is wrong :S
+            do_read(FHd, Filename, TaggedCsum, Offset, Size);
+        {false, {error, _}} ->
```

Comment:
Hereby a follow-up is needed in later pull requests. If a requested range includes any byte that is trimmed it should return as trimmed?